### PR TITLE
[CUDA_Runtime] update platform augmentation

### PR DESCRIPTION
**Pull request by simonbyrne**
Update CUDA_Runtime platform augmentation for #6449.

cc @maleadt

### CODE REVIEW

File: C/CUDA/CUDA_Runtime/build_tarballs.jl [hunk 1]
```diff
@@ -5,7 +5,7 @@ include(joinpath(YGGDRASIL_DIR, "fancy_toys.jl"))
 include(joinpath(YGGDRASIL_DIR, "platforms", "cuda.jl"))
 
 name = "CUDA_Runtime"
-version = v"0.4.0"
+version = v"0.5.0"
```
Comment (simonbyrne) [inline C/CUDA/CUDA_Runtime/build_tarballs.jl:8]: I'm not sure if this needs to be a minor release, or should it just be a patch release?

Comment (maleadt) [reply C/CUDA/CUDA_Runtime/build_tarballs.jl:8]: Let's just have it move in lockstep with CUDA_Driver_jll for now.

**Comment by simonbyrne** [comment]
@maleadt this is now ready for review.

File: C/CUDA/CUDA_Runtime/platform_augmentation.jl [hunk 2]
```diff
@@ -47,7 +47,7 @@ function cuda_toolkit_tag()
         cuda_driver = if CUDA_Driver_jll.is_available()
             @debug "Using CUDA_Driver_jll for driver discovery"
 
-            if !isdefined(CUDA_Driver_jll, :libcuda)
+            if isnothing(CUDA_Driver_jll.libcuda)
```
Comment (ancapdev) [inline C/CUDA/CUDA_Runtime/platform_augmentation.jl:50]: I'm not sure if this has affected more people, I assume it has, but my environment blew up with this change. I believe this code can be pulled in without the required `CUDA_Driver_jll` changes, which then hits an undefined variable error here.

Comment (maleadt) [reply C/CUDA/CUDA_Runtime/platform_augmentation.jl:50]: I don't see how that can happen; https://github.com/JuliaBinaryWrappers/CUDA_Runtime_jll.jl/blob/c255c7f1878a00d690d7eb138a656c1d196e38a6/Project.toml#L16

Comment (ancapdev) [reply C/CUDA/CUDA_Runtime/platform_augmentation.jl:50]: Beats me, all I know is it failed, and I don't use any of these packages directly, they're coming in via Flux, so I have no personal experience dealing with it. I fixed it by pinning it to 0.4.0, will look more into what may have happened to the environment state later.

**Pull request merged**